### PR TITLE
infra: Use PRs to compose release notes

### PR DESCRIPTION
**Pull request by qinqon**
**What this PR does / why we need it**:
Right now the release notes comes from the commits, but the info needed
is the release-note mark from PRs since users decide if they want that
info to appear that the release notes or not.

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -177,8 +177,10 @@ prepare-minor:
 prepare-major:
 	./hack/prepare-release.sh major
 
+release-notes:
```
Comment (phoracek) [inline Makefile:180]: Who calls this?

File: hack/release-notes.tmpl [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{{with .NotesWithActionRequired -}}
+# Urgent Upgrade Notes
```
Comment (phoracek) [inline hack/release-notes.tmpl:2]: Please include a markdown file describing how is this used. Ideally we would have RELEASE.md file in the root of the project describing how to handle release notes, how to create releases, ...

**Review by phoracek — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (qinqon) [reply Makefile:180]: This target is used by users to test release notes with something like

```bash
make release-notes WHAT="v0.65.1 v0.65.2"
```

This way to can check how they look before doing a release and if something is missing change the PR description is not ok.

File: RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+The release process for CNAO consist on a pair of manual steps:
+1. Create release PR with github action [Create Version PR](https://github.com/kubevirt/cluster-network-addons-operator/actions/workflows/prepare-version.yaml), 
+   it have a pair of fields `versionLevel` with means the semVer version part 
+   to bump and `baseBranch` the branch to relase from. To run it:
+   - From the gh UI passing the `versionLevel` or `baseBranch`
+   - With the gh cli like `gh workflow run prepare-version.yaml -s versionLevel=... -s baseBranch=...`
+2. Merge the version PR when continous integration is fine. 
+
+After the version PR is merged a prow job will [release-cluster-network-addons-operator](https://github.com/kubevirt/project-infra/blob/main/github/ci/prow-deploy/files/jobs/kubevirt/cluster-network-addons-operator/cluster-network-addons-operator-postsubmits.yaml#L2-L28) this will call the `make release` target that do the following:
```
Comment (rhrazdil) [inline RELEASE.md:11]: ~~this will~~ call the `make release` target **to** do the following

File: RELEASE.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Release process
+
+The release process for CNAO consist on a pair of manual steps:
```
Comment (rhrazdil) [inline RELEASE.md:3]: consists of

File: RELEASE.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+1. Create release PR with github action [Create Version PR](https://github.com/kubevirt/cluster-network-addons-operator/actions/workflows/prepare-version.yaml), 
+   it have a pair of fields `versionLevel` with means the semVer version part
```
Comment (rhrazdil) [inline RELEASE.md:5]: it has a pair of fields: `versionLevel` and `baseBranch`. `versionLevel` is the semVer version part to bump to and `baseBranch` is the branch to release from.

File: Makefile (hunk 1 above)
Comment (phoracek) [reply Makefile:180]: Understood

File: RELEASE.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+   it have a pair of fields `versionLevel` with means the semVer version part 
+   to bump and `baseBranch` the branch to relase from. To run it:
+   - From the gh UI passing the `versionLevel` or `baseBranch`
+   - With the gh cli like `gh workflow run prepare-version.yaml -s versionLevel=... -s baseBranch=...`
+2. Merge the version PR when continous integration is fine. 
+
+After the version PR is merged a prow job will [release-cluster-network-addons-operator](https://github.com/kubevirt/project-infra/blob/main/github/ci/prow-deploy/files/jobs/kubevirt/cluster-network-addons-operator/cluster-network-addons-operator-postsubmits.yaml#L2-L28) this will call the `make release` target that do the following:
+
+1. Tag the code the version from the version PR.
+2. Generate release notes from the PRs release-notes field at description
```
Comment (phoracek) [inline RELEASE.md:14]: I miss the information on what are the release notes generated from. How can I mark a PR to be included there, how can I mark it not to get listed, how can I mark it as "Urgent Upgrade Note".

**Review by phoracek — CHANGES_REQUESTED** [review]

File: RELEASE.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+
+## Release notes handling
+
+The release notes are generated using the kubernetes [release-notes](https://github.com/kubernetes/release/blob/master/cmd/release-notes/README.md).
+The tool scan all the PRs between the version to release and the previous one
+and look for the following block on the PR description:
+
+```
+\`\`\`release-note
```
Comment (phoracek) [inline RELEASE.md:29]: This does not render properly in the markdown output. I think you can drop the escaping

**Review by phoracek — CHANGES_REQUESTED** [review]
Small nits. The rest looks ok

**Review by phoracek — APPROVED** [review]
/lgtm

Thanks, this is great

**Review by rhrazdil — APPROVED** [review]
Thanks!

**Pull request merged**